spec/unit/stats/key_generator_spec.rb

```diff
@@ -0,0 +1,122 @@
+require_relative '../../spec_helper'
+
+RSpec.describe ThreeScale::Backend::Stats::KeyGenerator do
+  let(:service_id) { '123456' }
+  let(:applications) { %w[] }
+  let(:metrics) { %w[] }
+  let(:users) { %w[] }
+  let(:from) { Time.new(2002, 10, 31) }
+  let(:to) { Time.new(2002, 10, 31) }
+  let(:job_params) do
+    {
+      service_id: service_id,
+      applications: applications,
+      metrics: metrics,
+      users: users,
+      from: from.to_i,
+      to: to.to_i
+    }
+  end
+  subject { described_class.new(job_params).keys }
+
+  context 'responsecode_service keys' do
+    let(:expected_keys) do
+      %w[200 2XX 403 404 4XX 500 503 5XX].product(%i[hour day week month eternity]).map do |code, gr|
+        ThreeScale::Backend::Stats::Keys.service_response_code_value_key(service_id, code, ThreeScale::Backend::Period[gr].new(from))
```

Comment:
Would be good to have a comment somewhere explaining that these tests just check that a subset of the keys that are supposed to be generated are there.

Reply:
each context checks a subset of keys. That's what you mean?